pinot-controller/src/main/java/org/apache/pinot/controller/validation/RealtimeSegmentValidationManager.java

```diff
@@ -108,8 +112,34 @@ protected void processTable(String tableNameWithType, Context context) {
     if (context._runSegmentLevelValidation) {
       runSegmentLevelValidation(tableConfig, streamConfig);
     }
-    _llcRealtimeSegmentManager.ensureAllPartitionsConsuming(tableConfig, streamConfig,
-        context._recreateDeletedConsumingSegment, context._offsetCriteria);
+
+    // Keeps the table paused/unpaused based on storage quota validation.
+    // Skips updating the pause state if table is paused by admin
+    PauseState pauseState = computePauseState(tableNameWithType);
+
+    if (!pauseState.isPaused()) {
```

Comment:
Sorry for the nit, but this reads like it is invoking the periodic consumption fix if the table is not paused. That is partially true, but another factor is that the table may have run out of quota.

So, it is better to create a method like `shouldEnsurePartitionsConsuming()` from which we return a boolean instead of returning the PauseState.

Thanks.

Reply:
Had a chat offline, will handle the future of `_recreateDeletedConsumingSegment` flag in a separate PR.

Reply:
I learned that the flag is not quite as I had imagined. Yes, it is a config, and if turned true, then it ends up invoking the periodic job with 'true' (which ends up handling a condition of someone accidentally or intentionally removing the consuming segments of a table).

the flag is _also_ used in the "resume" part because resume happens via kicking off the periodic task.

cc: @sajjad-moradi